logstash-core/lib/logstash/pipeline.rb

```diff
@@ -107,16 +107,23 @@ def compile_lir
     LogStash::Compiler.compile_sources(sources_with_metadata, @settings)
   end
 
-  def plugin(plugin_type, name, *args)
+  def plugin(plugin_type, name, line, column, *args)
     @plugin_counter += 1
 
     # Collapse the array of arguments into a single merged hash
     args = args.reduce({}, &:merge)
 
-    id = if args["id"].nil? || args["id"].empty?
-      args["id"] = "#{@config_hash}-#{@plugin_counter}"
-    else
-      args["id"]
+    # Pull the ID from LIR to keep IDs consistent between the two representations
+    id = lir.graph.vertices.filter do |v| 
```

Comment:
@andrewvc I was actually hoping we could just not generate this but simply add the method that generates this ID to `config_ast.rb` :) but yea, it's doesn't make it much worse. Cool with me :) Feel free to merge once green :)